roc.go

```diff
@@ -0,0 +1,122 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+// ROC returns paired false positive rate (FPR) and true positive rate
+// (TPR) values corresponding to n cutoffs spanning the relative
+// (or receiver) operator characteristic (ROC) curve obtained when y is
+// treated as a binary classifier for classes with weights.
+//
+// Cutoffs are equally spaced from eps less than the minimum value of y
+// to the maximum value of y, including both endpoints -- this means the
+// resulting ROC curve will always begin at (0,0) and end at (1,1).
+//
+// The input y must be sorted, and SortWeightedLabeled can be used in
+// order to sort y together with classes and weights.
+//
+// For a given cutoff value, observations corresponding to entries in y
+// greater than the cutoff value are classified as false, while those
+// below (or equal to) the cutoff value are classified as true. These
+// assigned class labels are compared with the true values in the classes
+// slice and used to calculate the FPR and TPR.
+//
+// If weights is nil, all weights are treated as 1.
+//
+// n of zero results in all possible cutoffs being calculated -- this will
```

Comment:
Don't use `--` in comments. It has not markup meaning to godoc. I think this para could be rewritten to make it more English, since at the moment it reads more like code than prose.